Задача по теме "Написание методов JpaRepository"

Sindicat007 opened the pull request.

Sindicat007 committed:
Задача по теме "Написание методов JpaRepository"

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

RinatBeybutov commented on JpaRepository/src/main/java/ru/maxima/jpa/repository/PostRepository.java:
```diff
@@ -0,0 +1,21 @@
+
+    List<PostEntity> findPostByUser(UserEntity userEntity, Pageable pageable);
+
+    @Query(value = "SELECT * FROM users.posts WHERE title LIKE %:title%", nativeQuery = true)
```
давай перепишем на запрос с jpql.
Проблема n+1 не решена

RinatBeybutov commented on JpaRepository/src/main/java/ru/maxima/jpa/repository/UserRepository.java:
```diff
@@ -0,0 +1,19 @@
+    List<UserEntity> findByUsername(String username);
+
+    @Query(value = "SELECT * FROM users.user WHERE age > :age", nativeQuery = true)
+    List<UserEntity> findAllByAgeGreaterThanCustom(@Param("age") Integer age);
```
давай перепишем на запрос с jpql

RinatBeybutov commented on JpaRepository/src/main/java/ru/maxima/jpa/service/PostService.java:
```diff
@@ -0,0 +1,70 @@
+
+        PostEntity postEntity = new PostEntity();
+        postEntity.setUser(user);
+        postEntity.setTitle(postMapper.toEntity(postDTO).getTitle());
```
мапперы не так используются, после метода маппера получается PostEntity, который ты наполняешь нужными полями (типо user)

RinatBeybutov commented on JpaRepository/src/main/java/ru/maxima/jpa/entity/UserEntity.java:
```diff
@@ -0,0 +1,22 @@
+import lombok.Setter;
+
+@Entity
+@Table(schema = "users", name = "user")
```
таблица должна быть users, а схема другая. Можно в тестовом задании использовать public, user_service или что-то другое

RinatBeybutov commented on JpaRepository/src/main/java/ru/maxima/jpa/entity/PostEntity.java:
```diff
@@ -0,0 +1,34 @@
+import java.time.LocalDateTime;
+
+@Entity
+@Table(schema = "users", name = "posts")
```
поправить схему

RinatBeybutov commented on JpaRepository/src/main/java/ru/maxima/jpa/repository/UserRepository.java:
```diff
@@ -0,0 +1,19 @@
+import ru.maxima.jpa.entity.UserEntity;
+
+import java.util.List;
+
```
давай добавим над каждым классом javaDocs:
```
/*
* Репозиторий для работы с пользователями
* @author Alexey
*/
```

RinatBeybutov commented on JpaRepository/src/main/java/ru/maxima/jpa/Main.java:
```diff
@@ -0,0 +1,44 @@
+
+public class Main {
+    public static void main(String[] args) {
+        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
```
давай это перепишем на springBoot, а вот этот код перенесем в тесты с использованием тестконтейнерс

RinatBeybutov commented on JpaRepository/src/main/java/ru/maxima/jpa/service/UserService.java:
```diff
@@ -0,0 +1,43 @@
+
+    @Transactional
+    public void createUser(UserDTO userDTO) {
+        userRepository.save(userMapper.toEntity(userDTO));
```
нужно разнести на несколько строчек, а еще обычно метод создания и обновления возвращает новый объект

Sindicat007 committed:
- Написал тесты для репозиториев
- Переписал приложение на SpringBoot
- Переписал кастомные запросы в репозиториях с jpql
- Решил проблему n + 1 в PostRepository
- Поправил использование маппера в PostService
- Поправил название схем в сущностях
- Добавил javaDocs над классами

Signed-off-by: Sindicat007 <sindicat007@ya.ru>
